modin/experimental/engines/omnisci_on_ray/frame/data.py

```diff
@@ -1055,11 +1053,24 @@ def reset_index(self, drop):
                 raise NotImplementedError(
                     "default index reset with no drop is not supported"
                 )
-            new_columns = Index.__new__(Index, data=self._table_cols, dtype="O")
+            # Need to demangle index names.
+            exprs = OrderedDict()
+            for i, c in enumerate(self._index_cols):
+                name = self._index_name(c)
+                if name is None:
+                    name = f"level_{i}"
+                if name in exprs:
+                    raise ValueError(f"cannot insert {name}, already exists")
+                exprs[name] = self.ref(c)
+            for c in self.columns:
+                if c in exprs:
+                    raise ValueError(f"cannot insert {c}, already exists")
```

Comment:
can we add a test, that will hit that error? For now, this case is uncovered by tests

Reply:
we have a function [`eval_general`](https://github.com/modin-project/modin/blob/48d0e0c2436fe4064b611ad412b686701f104535/modin/pandas/test/utils.py#L634-L642) for handy testing at `utils.py`, this function encapsulates exception checking, so tests could look like this:
```python
eval_general(md_df, pd_df, lambda df, *args, **kwargs: df.reset_index(*args, **kwargs))
```

Reply:
ok, added these cases